clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11724,6 +11724,9 @@ def err_sycl_restrict : Error<
   "|use a const static or global variable that is neither zero-initialized "
   "nor constant-initialized"
   "}0">;
+def err_sycl_external_global : Error<
+  "SYCL device code cannot reference an external device_global variable not "
```

Comment:
I'm not a huge fan of this diagnostic. WDYT about 

`invalid reference to device_global variable; external device_global variable must be marked with SYCL_EXTERNAL macro `

@premanandrao thoughts?

Reply:
LGTM, I've changed it.

Reply:
I like it.  I would also suggest:
``invalid reference to external 'device_global' variable; such variables must be marked with SYCL_EXTERNAL macro``

Reply:
I'm not a huge fan of 'such variables'. I think  `invalid reference to external 'device_global' variable; variable must be marked with SYCL_EXTERNAL macro` suffices. I don't mind the existing diagnostic as well.

Reply:
Fine by me.